packages/task/src/browser/task-service.ts

```diff
@@ -88,12 +92,23 @@ export class TaskService implements TaskConfigurationClient {
     @inject(EditorManager)
     protected readonly editorManager: EditorManager;
 
+    @inject(ProblemManager)
+    protected readonly problemManager: ProblemManager;
+
+    @inject(ProblemMatcherRegistry)
+    protected readonly problemMatcherRegistry: ProblemMatcherRegistry;
+
+    @inject(TaskDefinitionRegistry)
+    protected readonly taskDefinitionRegistry: TaskDefinitionRegistry;
+
     /**
      * @deprecated To be removed in 0.5.0
      */
     @inject(TaskProviderRegistry)
     protected readonly taskProviderRegistry: TaskProviderRegistry;
 
+    private problemMatchers: string[] = [];
```

Comment:
this is a hack. will use the function in TaskClient to parse output and add markers.